appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/CommitDelayingResponse.java

```diff
@@ -77,9 +77,6 @@
    */
   public CommitDelayingResponse(HttpServletResponse response) throws IOException {
     super(response);
-    // Set the buffer size explicitly, as this ensure both that jetty's aggregate
-    // buffer is the correct size and that the commit threshold is that size.
-    response.setBufferSize(CommitDelayingOutputStream.MAX_RESPONSE_SIZE_BYTES);
```

Comment:
Does Jetty use a pool of direct byte buffers for the responses? When we set the buffer to 32MB, how much off-heap memory can it potentially use up?

The hard fail at 32MB is fine because that mimics the behavior of Standard. I do like the idea of modifying the output stream wrapper to hold the response in a growing buffer up to 32MB. I believe that was the original intent, but wasn't correctly implemented. Also, the buffer should be growing and allocated on the heap.